src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
+
+  return sum / women.length;
```

Comment:
DRY rule: You can create function, which will contain this code and can be reused for `calculateWomenAverageAge` function and `calculateMenAverageAge` , for example:

```javascript
function calculateAverageAge(people) {
  const totalAge = people.reduce((sum, person) => sum + (person.died - person.born), 0);

  return totalAge / people.length;
}
```

And you can use it this way (`calculateMenAverageAge` example):
```javascript
function calculateMenAverageAge(people, century) {
  const men = century
    ? people.filter(person => person.sex === 'm' && Math.ceil(person.died / 100) === century)
    : people.filter(person => person.sex === 'm');

  return calculateAverageAge(men);
}
```